externs/es6.js

```diff
@@ -330,41 +330,228 @@ var BufferSource;
+TypedArray.prototype.length;
 
-/** @const {number} */
-ITypedArray.prototype.BYTES_PER_ELEMENT;
+/**
+ * @param {function(this:S, number, number, !TypedArray) : number} callback
+ * @param {S=} opt_thisArg
+ * @return {!TypedArray}
+ * @template S
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/map
+ */
+TypedArray.prototype.map = function(callback, opt_thisArg) {};
+
+/**
+ * @param {function((number|INIT|RET), number, number, !TypedArray) : RET} callback
+ * @param {INIT=} opt_initialValue
+ * @return {RET}
+ * @template INIT,RET
+ * @nosideeffects
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/reduce
+ */
+TypedArray.prototype.reduce = function(callback, opt_initialValue) {};
 
 /**
- * @param {ArrayBufferView|Array<number>} array
+ * @param {function((number|INIT|RET), number, number, !TypedArray) : RET} callback
+ * @param {INIT=} opt_initialValue
+ * @return {RET}
+ * @template INIT,RET
+ * @nosideeffects
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/reduceRight
+ */
+TypedArray.prototype.reduceRight = function(callback, opt_initialValue) {};
+
+/**
+ * @return {!TypedArray}
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/reverse
+ */
+TypedArray.prototype.reverse = function() {};
+
+/**
+ * @param {!ArrayBufferView|!Array<number>} array
  * @param {number=} opt_offset
  * @return {undefined}
+ * @throws {!RangeError}
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/set
  */
-ITypedArray.prototype.set = function(array, opt_offset) {};
+TypedArray.prototype.set = function(array, opt_offset) {};
 
 /**
- * @param {number} begin
+ * @param {number=} opt_begin
  * @param {number=} opt_end
- * @return {ITypedArray}
+ * @return {!TypedArray}
  * @nosideeffects
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice
  */
-ITypedArray.prototype.subarray = function(begin, opt_end) {};
+TypedArray.prototype.slice = function(opt_begin, opt_end) {};
 
 /**
- * @param {number} value
- * @param {number=} opt_begin
+ * @param {function(this:S, number, number, !TypedArray) : boolean} callback
+ * @param {S=} opt_thisArg
+ * @return {boolean}
+ * @template S
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/some
+ */
+TypedArray.prototype.some = function(callback, opt_thisArg) {};
+
+/**
+ * @param {(function(number, number) : number)=} opt_compareFunction
+ * @return {!TypedArray}
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/sort
+ */
+TypedArray.prototype.sort = function(opt_compareFunction) {};
+
+/**
+ * @param {number} begin
  * @param {number=} opt_end
- * @return {ITypedArray}
+ * @return {!TypedArray}
```

Comment:
To get Closure library to compile, I had to change this to

```
 * @return {THIS}
 * @this {THIS}
 * @template THIS
```

I assume the same would be correct for all methods that return a TypedArray?